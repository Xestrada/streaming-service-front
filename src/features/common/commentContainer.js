import React from 'react';
import PropTypes from 'prop-types';
import './commentContainer.scss';

class CommentContainer extends React.Component {

    static propTypes = {
        comment: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        buttonText: PropTypes.string.isRequired,
        placeHolderText: PropTypes.shape.isRequired,
        buttonFunc: PropTypes.func.isRequired,
        changeFunc: PropTypes.func.isRequired,
    };

    render() {
        const { title, buttonText, placeHolderText, buttonFunc, changeFunc, comment } = this.props;

        const commentContainer = (
            <div id='comment-container'>
                <div id='comment-header'>
                    <label htmlFor='Comment' style={{ textDecoration: 'underline', fontFamily: 'Apple Chancery, cursive' }}>
                        { title }
                    </label>
                </div>
                <textarea

                  onChange={e => changeFunc(e.target.value)} //eslint-disable-line
                  id='subject' //eslint-disable-line
                  name='subject' //eslint-disable-line
                  value={comment} //eslint-disable-line
                  placeholder={placeHolderText} //eslint-disable-line
                  style={{ borderStyle: 'inset', width: '600px', height: '90px' }} //eslint-disable-line
                />
                <div className='row'>
                    <input style={{ marginLeft: '52%' }} type='submit' value={buttonText} onClick={buttonFunc} />
                </div>
            </div>
        );

        return (
            <div>
                { commentContainer }
            </div>
        );
    }

}

export default CommentContainer;
