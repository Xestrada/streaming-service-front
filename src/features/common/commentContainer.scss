.comment-container {

    width: 100%;
    height: 23%;
    padding: 20px;

    .comment-label {
        display: inline;
        color: whitesmoke;
        font-size: 1em;
        text-decoration: underline;
        margin-right: 0.5rem;
    }

    .textArea {
        display: inline;
        border-style: 'inset';
        width: 20em;
    }

    .postButton {
        display: inline;
        margin-left: 0.5rem;
    }

}