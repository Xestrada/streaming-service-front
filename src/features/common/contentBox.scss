.contentBox{
    margin: 0;
    display: inline-block;
    .imgDiv{
        display: block;
        width: 8em;
    }
    .centerText {
        margin: auto;
        text-align: center;
    }

}