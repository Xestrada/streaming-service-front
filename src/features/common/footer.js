import React from 'react';
import {
    Media,
} from 'reactstrap';
import { Link } from 'react-router-dom';
import facebook from '../../images/facebooklogo.png';
import twitter from '../../images/twitterlogo.jpg';
import googleplus from '../../images/googlepluslogo.png';
import './footer.scss';


export default class Footer extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isOpen: false,
        };
        this.toggle = this.toggle.bind(this);
    }

    toggle() {
        this.setState(prevState => ({
            isOpen: !prevState.isOpen,
        }));
    }

    render() {
        return (
            <div className='dark'>
                <h1 className='relative'>Contact us at videovault.company48@gmail.com</h1>
                <div className='d-flex justify-content-center'>
                    <Media href='https://facebook.com'>
                        <Media object src={facebook} className='img' alt='Generic placeholder image' />
                    </Media>
                    <Media href='https://twitter.com'>
                        <Media object src={twitter} className='img' alt='Generic placeholder image' />
                    </Media>
                    <Media href='https://google.com'>
                        <Media object src={googleplus} className='img' alt='Generic placeholder image' />
                    </Media>
                </div>
                <div className='relative'>
                    <Link to='/privacy'>Privacy Policy</Link>
                </div>

            </div>
        );
    }
}
