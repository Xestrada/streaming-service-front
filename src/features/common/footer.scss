
.img {
        height: 75px;
        width: 150px;
        padding-right: 50px;
        padding-left: 50px;
        padding-top: 25px;
    }

.relative {
    position: relative;
    text-align: center;
    font-size: 16px;

}
div.dark{
    background-color: rgba(0, 0, 0, 0.24);
}
