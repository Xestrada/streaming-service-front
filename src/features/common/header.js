import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import {
    Alert,
    Collapse,
    Dropdown,
    DropdownItem,
    DropdownMenu,
    DropdownToggle,
    Navbar,
    NavbarToggler,
    Nav,
    NavItem,
    Button,
    Media,
    Modal,
    ModalHeader,
    ModalBody,
    ModalFooter,
} from 'reactstrap';
import { Link, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import * as actions from './redux/actions';
import userImg from '../../images/blank-user.jpg';
import './header.scss';

class Header extends React.Component {
    static propTypes = {
        actions: PropTypes.object.isRequired,
        common: PropTypes.object.isRequired,
        location: PropTypes.object.isRequired,
    };

    constructor(props) {
        super(props);

        this.state = {
            isOpen: false,
            modal: false,
            userDropdown: false,
            username: '',
            pass: '',
            needToSub: false,
        };

        this.navToggle = this.navToggle.bind(this);
        this.modalToggle = this.modalToggle.bind(this);
        this.dropdownToggle = this.dropdownToggle.bind(this);
        this.login = this.login.bind(this);
        this.signOut = this.signOut.bind(this);
        this.updateState = this.updateState.bind(this);
    }

    componentDidMount() {
        const { actions } = this.props;
        const { localAuthen } = actions;
        const id = localStorage.getItem('id');
        const username = localStorage.getItem('username');
        const email = localStorage.getItem('email');
        if (id !== null && username !== null) {
            localAuthen({
                id,
                username,
                email,
            });
        }
    }

    dropdownToggle() {
        this.setState(prevState => ({
            userDropdown: !prevState.userDropdown,
        }));
    }

    navToggle() {
        this.setState(prevState => ({
            isOpen: !prevState.isOpen,
        }));
    }

    modalToggle() {
        this.setState(prevState => ({
            modal: !prevState.modal,
        }));
    }

    login() {
        const { actions } = this.props;
        const { authen } = actions;
        const { username, pass } = this.state;
        authen(username, pass).then(() => {
            const { common, actions } = this.props;
            const { authen, userData } = common;
            const { hasAllSlots } = actions;
            if (authen) {
                hasAllSlots(userData.id).then(() => {
                    const { common } = this.props;
                    const { areSlotsFull } = common;
                    this.setState({
                        needToSub: !areSlotsFull,
                    });
                });
            }
        });

    }

    signOut() {
        const { actions } = this.props;
        const { signOut } = actions;
        signOut();
    }

    updateState(name, value) {
        this.setState({
            [name]: value,
        });
    }

    render() {
        const { modal, username, pass, isOpen, userDropdown, needToSub } = this.state;
        const { common, location } = this.props;
        const { authen, authenError, authenPending, userData } = common;
        const { pathname } = location;

        if (authen && modal) this.modalToggle();

        if (needToSub && pathname !== '/sub-init') {
            return (<Redirect to='/sub-init' />);
        }


        const errorElem = authenError !== null ? (
            <div>
                {authenError.invalid_email ? (<Alert color='danger'>Invalid Email</Alert>) : null}
                {authenError.invalid_password ? <Alert color='danger'>Invalid Password</Alert> : null}
            </div>
        ) : null;

        const modalElem = (
            <Modal isOpen={modal} toggle={this.modalToggle}>
                <ModalHeader toggle={this.modalToggle} className='centerModalHeader'>Member Login</ModalHeader>
                <ModalBody className='modalBody'>
                    {errorElem}
                    <input value={username} type='text' id='userName' className='form-control' placeholder='email' onChange={e => this.updateState('username', e.target.value)} />
                    <input value={pass} type='password' id='userPassword' className='form-control input-sm chat-input' placeholder='password' onChange={e => this.updateState('pass', e.target.value)} />
                </ModalBody>
                <ModalFooter>
                    {authenPending && <i className='fa fa-spinner fa-spin loadIcon' />}
                    <Button className='btn btn-primary btn-md' color='primary' onClick={this.login}>
                        login
                        <i className='fas fa-sign-in-alt' />
                    </Button>
                    <Button color='secondary' onClick={this.modalToggle}>Cancel</Button>
                </ModalFooter>
            </Modal>
        );

        const sideHead = authen ? (
            <Dropdown isOpen={userDropdown} toggle={this.dropdownToggle} direction='down'>
                <DropdownToggle>
                    <Media right>
                        <img className='user' src={userImg} alt='Placeholder' />
                    </Media>
                </DropdownToggle>
                <DropdownMenu right>
                    <DropdownItem><Link to='/account'>Account</Link></DropdownItem>
                    <DropdownItem onClick={this.signOut}>Sign Out</DropdownItem>
                </DropdownMenu>
            </Dropdown>
        ) : (
            <div>
                <Button className='color-me link' color='white' onClick={this.modalToggle}>
                Login
                </Button>

                <Link className='color-me link' to='/signup'>Sign Up</Link>
            </div>
        );

        return (
            <div>

                {modalElem}

                <Navbar color='dark' light expand='md'>

                    <Link className='color-me brand' to='/'>Company 48</Link>

                    <NavbarToggler onClick={this.navToggle} />

                    <Collapse isOpen={isOpen} navbar>

                        <Nav className='mr-auto' navbar>

                            <NavItem>
                                <Link className='color-me link' to='/movies'>Movies</Link>
                            </NavItem>

                            <NavItem>
                                <Link className='color-me link' to='/tvshows'>TV Shows</Link>
                            </NavItem>
                            {authen ? (
                                <NavItem>
                                    <Link className='color-me link' to='/subscriptions'>Subscriptions</Link>
                                </NavItem>
                            )
                                : null}

                            {authen ? (
                                <NavItem>
                                    <Link className='color-me link' to='/profile'>Profile</Link>
                                </NavItem>
                            )
                                : null}


                            <NavItem>
                                <Link className='color-me link' to='/about'>About</Link>
                            </NavItem>

                        </Nav>
                        <Nav className='spacing' />
                        <Nav className='ml-auto' navbar>
                            <NavItem>
                                <Link className='color-me link username' to='/subscriptions'>{userData === undefined || userData === null ? '' : userData.username}</Link>
                            </NavItem>
                            {sideHead}

                        </Nav>

                    </Collapse>

                </Navbar>
            </div>
        );
    }
}

/* istanbul ignore next */
function mapStateToProps(state) {
    return {
        header: state.header,
        common: state.common,
    };
}

/* istanbul ignore next */
function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({ ...actions }, dispatch),
    };
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Header));
