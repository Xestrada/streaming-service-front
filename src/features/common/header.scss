.color-me{
    color: white !important;
}
.centerModalHeader{
    margin-left: 4em;
}
.modalBody{
    margin-right: 5em ;
    margin-left: 5em;
}
 
.container {
position: fixed;
}

.spacing{
justify-content: space-between;
text-align: center;
margin-top: 0.18em;
margin-right: 2em;
}
.remember{
font-size: 15px;
margin: 0 5px;
}
