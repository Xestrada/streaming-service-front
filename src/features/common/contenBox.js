import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './contentBox.scss';

export default class ContentBox extends React.Component {

  static propTypes = {
      image: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
  };


  render() {

      const { image, title } = this.props;

      return (
          <div className='contentBox'>
              <img src={image} alt='default' className='imgDiv' />
              <Link to={
                  {
                      pathname: `/media/${title}`,
                      state: {
                          title,
                      },
                  }
              }
              >
                  <h6 className='centerText'>{title}</h6>
              </Link>
          </div>
      );
  }
}
