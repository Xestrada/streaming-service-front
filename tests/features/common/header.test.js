import React from 'react';
import { Header } from '../../../src/features/common';


describe('common/header', () => {
  it('renders correctly', () => {
    const renderedComponent =<Header />;
    expect(renderedComponent).toMatchSnapshot();
  });
});
