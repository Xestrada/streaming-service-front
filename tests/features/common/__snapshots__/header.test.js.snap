// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/header renders correctly 1`] = `<Unknown />`;
